ts/main.ts

```diff
@@ -44,6 +42,88 @@ $form.addEventListener('submit', (event: Event) => {
   };
   data.nextEntryId++;
   data.entries.unshift(formData);
+
+  $ul.prepend(renderEntry(formData));
+  viewSwap('entries');
+  toggleNoEntries();
   $image.src = 'images/placeholder-image-square.jpg';
   $form.reset();
 });
+
+function renderEntry(entry: DataValues): HTMLLIElement {
+  const $li = document.createElement('li');
+  $li.setAttribute('class', 'row');
+
+  const $entryImage = document.createElement('div');
+  $entryImage.setAttribute('class', 'column-half view-entry');
+
+  const $viewUrl = document.createElement('img');
+  $viewUrl.setAttribute('src', entry.newUrl);
+
+  const $entryContent = document.createElement('div');
+  $entryContent.setAttribute('class', 'column-half');
+
+  const $viewTitle = document.createElement('h2');
+  $viewTitle.textContent = entry.title;
+
+  const $viewNotes = document.createElement('p');
+  $viewNotes.textContent = entry.notes;
+
+  $li.appendChild($entryImage);
+  $entryImage.appendChild($viewUrl);
+  $li.appendChild($entryContent);
+  $entryContent.appendChild($viewTitle);
+  $entryContent.appendChild($viewNotes);
+
+  return $li;
+}
+
+document.addEventListener('DOMContentLoaded', () => {
+  for (let i = 0; i < data.entries.length; i++) {
+    $ul.appendChild(renderEntry(data.entries[i]));
+  }
+  viewSwap(data.view);
+  toggleNoEntries();
+});
+
+const $toggle = document.querySelector('#no-entries');
+if (!$toggle) throw new Error('$toggle query failed');
+
+function toggleNoEntries(): void {
+  if (!$toggle) throw new Error('$toggle query failed');
+  if (data.entries.length !== 0) {
+    $toggle.className = 'hidden';
+  }
```

Comment:
You also need to handle the opposite. If there are no entries, then we want the text to show again